src/SignalR/clients/csharp/Client.SourceGenerator/src/HubProxyGenerator.Parser.cs

```diff
@@ -0,0 +1,345 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.AspNetCore.SignalR.Client.SourceGenerator
+{
+    internal partial class HubProxyGenerator
+    {
+        internal class Parser
+        {
+            internal static bool IsSyntaxTargetForAttribute(SyntaxNode node) => node is AttributeSyntax
+            {
+                Name: IdentifierNameSyntax
+                {
+                    Identifier:
+                    {
+                        Text: "GetProxy"
+                    }
+                },
+                Parent:
+                {
+                    Parent: MethodDeclarationSyntax
+                    {
+                        Parent: ClassDeclarationSyntax
+                    }
+                }
+            };
+
+            internal static MethodDeclarationSyntax? GetSemanticTargetForAttribute(GeneratorSyntaxContext context)
+            {
+                var attributeSyntax = (AttributeSyntax)context.Node;
+                var attributeSymbol = context.SemanticModel.GetSymbolInfo(attributeSyntax).Symbol;
+
+                if (attributeSymbol is null ||
+                    attributeSymbol.ToString() != "Microsoft.AspNetCore.SignalR.Client.GetProxyAttribute.GetProxyAttribute()")
```

Comment:
I don't remember why I spelled it out as such so I'll change it.

Reply:
So this one is a bit confusing.. `typeof(HubServerProxyAttribute)` raises an exception during build of unit test project. (attached debugger to build process) Not exactly sure what's wrong since generator itself builds fine and intellisense works OK in `HubServerProxyGenerator.cs` as well as `HubServerProxyGenerator.Parser.cs`.

```
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.AspNetCore.SignalR.Client.Core, Version=42.42.42.42, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.
File name: 'Microsoft.AspNetCore.SignalR.Client.Core, Version=42.42.42.42, Culture=neutral, PublicKeyToken=adb9793829ddae60'
   at Microsoft.AspNetCore.SignalR.Client.SourceGenerator.HubServerProxyGenerator.Parser.GetSemanticTargetForGeneration(GeneratorSyntaxContext context)
   at Microsoft.AspNetCore.SignalR.Client.SourceGenerator.HubServerProxyGenerator.<>c.<Initialize>b__1_4(GeneratorSyntaxContext ctx, CancellationToken _)
   at Microsoft.CodeAnalysis.UserFunctionExtensions.<>c__DisplayClass0_0`2.<WrapUserFunction>b__0(TInput input, CancellationToken token)
```